Code/Mantid/MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -9565,7 +9758,11 @@ void ApplicationWindow::savedProject()
     foreach(MdiSubWindow *w, folderWindows)
     {
       if (w->isA("Matrix"))
-        (dynamic_cast<Matrix*>(w))->undoStack()->setClean();
+      {
+        Matrix* m = dynamic_cast<Matrix*>(w);
+        if (m)
```

Comment:
Maybe if(m && m->undoStack()) for better safety? Not sure if we can trust the undo stack pointer in the Matrix class.